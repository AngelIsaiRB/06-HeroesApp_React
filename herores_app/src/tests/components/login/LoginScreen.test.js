import React from 'react';
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('pruebas en <LoginScreen/>', () => {

    const history = {
        length:2,
        push: jest.fn(),
        goBack: jest.fn(),
        replace: jest.fn(),
    }

    const contextValue={
        dispatch: jest.fn(),
        logged:false,
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history}></LoginScreen>
        </AuthContext.Provider>        
    );
    test('should de hacer mostrarse cor rectamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should de hacer el login', () => {
       const handleclick= wrapper.find("button").prop("onClick");
       handleclick();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: "isai"
            }
        });
        expect(history.replace).toHaveBeenCalledWith("/");
        
        localStorage.setItem("lastPath","/dc");
        handleclick();
        expect(history.replace).toHaveBeenCalledWith("/dc");

    });
    
    
    
})
