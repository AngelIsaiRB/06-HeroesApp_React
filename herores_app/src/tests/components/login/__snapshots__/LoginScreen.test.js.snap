// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <LoginScreen/> should de hacer mostrarse cor rectamente 1`] = `
<div
  clasname="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
